sem/arch/x86.py

```diff
@@ -106,30 +106,40 @@ def size_of(arg_type: str):
             # arguments to fill all SSE registers.
             if arg_type[0] not in ["i", "p", "u"]:
                 continue
-            if unused_gpr < len(arg_gprs):
-                # NOTE: Also assuming that i<size> is within word size
-                arg_gprs[unused_gpr].attr |= VarAttr.FUNCTION_ARG
-                if arg_type[0] == "p":
-                    arg_gprs[unused_gpr].attr |= VarAttr.PTR
-                    # ceil arg_type bit size to byte size
-                    heap_vars.append(
-                        RandMemVar(arg_gprs[unused_gpr], size_of(arg_type), self)
-                    )
-                elif arg_attr == "s":
-                    arg_gprs[unused_gpr] = SExtRegister(
-                        arg_gprs[unused_gpr]._reg, size_of(arg_type), self
-                    )
-                elif arg_attr == "z":
-                    arg_gprs[unused_gpr] = ZExtRegister(
-                        arg_gprs[unused_gpr]._reg, size_of(arg_type), self
-                    )
-                unused_gpr += 1
-            else:
-                stack_var = self._make_stack_arg(int(self._mode))
-                stack_vars.append(stack_var)
-                if arg_type[0] == "p":
-                    stack_var.attr |= VarAttr.PTR
-                    heap_vars.append(RandMemVar(stack_var, size_of(arg_type), self))
+            # NOTE: Also assuming that i<size> is within word size
+            arg_gprs[unused_gpr].attr |= VarAttr.FUNCTION_ARG
+            if arg_type[0] == "p":
+                arg_gprs[unused_gpr].attr |= VarAttr.PTR
```

Comment:
These changes should make the first extra parameter stored at the lowest address and the last extra parameter at the highest and so on.